packages/zoe/README.md

```diff
@@ -124,3 +132,25 @@ harden(prepare);
 ```
 
 For an example contract upgrade, see the test at https://github.com/Agoric/agoric-sdk/blob/master/packages/zoe/test/swingsetTests/upgradeCoveredCall/test-coveredCall-service-upgrade.js .
+
+### Durability
+
+The contract must retain in durable storage anything that must persist between incarnations. All other state will be lost.
+
+### Kinds
+
+The contract defines the kinds that are held in durable storage. Thus the function calls that define the kinds must be run before the objects are deserialized from durable storage.
+
+# Crank
+
+After the first incarnation, `prepare` must return in one crank. Therefore any
+remote calls must only be during the initial incaranation. During subsequent
+upgrades, `prepare` should be using a value that it stashed away in the baggage.
+The `provideAll` function in contract support is designed to help.
```

Comment:
Do we allow the prepare for an initial incarnation to not settle in the first crank? If not then this is inaccurate, but if so then it can be clarified:
```suggestion
For the first incarnation, `prepare` is allowed to return a promise that takes more than one crank to settle
(e.g., because it depends upon the results of remote calls).
But in later incarnations, `prepare` must settle in one crank.
Therefore such necessary values should be stashed in the baggage by earlier incarnations.
The `provideAll` function in contract support is designed to support this.
```